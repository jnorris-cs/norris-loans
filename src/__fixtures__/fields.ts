import type { FieldMetadata } from '../types';

import fields from '../../public/json/fieldsMetaData.json';

export default fields as FieldMetadata[];
