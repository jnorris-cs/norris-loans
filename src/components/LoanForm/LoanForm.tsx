import type { FieldMetadata } from 'types';

import LoanEntity from 'components/LoanEntity/LoanEntity';
import { UpdateLoanContext } from 'contexts/UpdateLoanContext';

import { useGroupFieldsByEntity } from './useGroupFieldsByEntity/useGroupFieldsByEntity';
import { useLoan } from './useLoan/useLoan';

interface LoanFormProps {
	fields: FieldMetadata[];
}

const LoanForm = ({ fields }: LoanFormProps) => {
	const entities = useGroupFieldsByEntity(fields); // convert flat fields into entity objects
	const updateLoanContext = useLoan();

	return (
		<form
			action="#"
			autoComplete="off"
			className="content pt3"
		>
			<UpdateLoanContext.Provider value={updateLoanContext}>
				{entities.map((entity) => (
					<LoanEntity
						entity={entity}
						key={entity.name}
					/>
				))}
			</UpdateLoanContext.Provider>
		</form>
	);
};

export default LoanForm;
