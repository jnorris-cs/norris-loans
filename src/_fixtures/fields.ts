import fields from "../../public/json/fieldsMetaData.json";
import type { FieldMetadata } from "../types";

export default fields as FieldMetadata[];
