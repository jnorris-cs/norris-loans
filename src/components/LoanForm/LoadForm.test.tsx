import { render } from '@testing-library/react';
import LoanForm from './LoanForm';
import fields from '../../_fixtures/fields';

describe('LoanForm', () => {
	it('renders the component correctly', () => {
		const result = render(<LoanForm fields={fields} />);
		expect(result.baseElement.firstChild).toMatchSnapshot();
	});
});
