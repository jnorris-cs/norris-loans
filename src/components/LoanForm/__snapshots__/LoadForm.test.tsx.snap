// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoanForm renders the component correctly 1`] = `
<div>
  <div
    class="content pt3"
  >
    <fieldset
      class="bw0 pa0 mb4"
    >
      <legend
        class="mb2"
      >
        <span
          class="fw6 f3 mr2"
        >
          Loan
        </span>
        <button
          class="pa0 bw0 bg-transparent"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-chevron-down "
            data-icon="chevron-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
              fill="currentColor"
            />
          </svg>
        </button>
      </legend>
      <div
        class="mb3 loan-field loan-field__valid"
      >
        <label
          class="db mb1"
          for="Loan.loanAmount"
        >
          Loan Amount
        </label>
        <input
          id="Loan.loanAmount"
          inputmode="decimal"
          name="Loan.loanAmount"
          type="text"
          value=""
        />
      </div>
      <div
        class="mb3 loan-field loan-field__valid"
      >
        <label
          class="db mb1"
          for="Loan.downPaymentAmount"
        >
          Down Payment
        </label>
        <input
          id="Loan.downPaymentAmount"
          inputmode="decimal"
          name="Loan.downPaymentAmount"
          type="text"
          value=""
        />
      </div>
    </fieldset>
    <fieldset
      class="bw0 pa0 mb4"
    >
      <legend
        class="mb2"
      >
        <span
          class="fw6 f3 mr2"
        >
          Borrower
        </span>
        <button
          class="pa0 bw0 bg-transparent"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-chevron-down "
            data-icon="chevron-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M233.4 406.6c12.5 12.5 32.8 12.5 45.3 0l192-192c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L256 338.7 86.6 169.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l192 192z"
              fill="currentColor"
            />
          </svg>
        </button>
      </legend>
      <div
        class="mb3 loan-field loan-field__valid"
      >
        <label
          class="db mb1"
          for="Borrower.firstName"
        >
          First Name
        </label>
        <input
          id="Borrower.firstName"
          name="Borrower.firstName"
          type="text"
          value=""
        />
      </div>
      <div
        class="mb3 loan-field loan-field__valid"
      >
        <label
          class="db mb1"
          for="Borrower.lastName"
        >
          Last Name
        </label>
        <input
          id="Borrower.lastName"
          name="Borrower.lastName"
          type="text"
          value=""
        />
      </div>
      <div
        class="mb3 loan-field loan-field__valid"
      >
        <label
          class="db mb1"
          for="Borrower.birthDate"
        >
          Birth Date
        </label>
        <input
          field="[object Object]"
          id="Borrower.birthDate"
          name="Borrower.birthDate"
          type="date"
          value=""
        />
      </div>
    </fieldset>
  </div>
</div>
`;
